client/my-sites/themes/theme-options.js

```diff
@@ -59,7 +59,8 @@ const activate = {
 const activateOnJetpack = {
 	label: i18n.translate( 'Activate' ),
 	header: i18n.translate( 'Activate on:', { comment: 'label for selecting a site on which to activate a theme' } ),
-	action: activateWpcomThemeOnJetpack,
+	// Append `-wpcom` suffix to the theme ID so the installAndActivate() will install the theme from WordPress.com, not WordPress.org
+	action: ( themeId, siteId, ...args ) => installAndActivate( themeId + '-wpcom', siteId, ...args ),
```

Comment:
True. I think I like to make `siteId` explicit since apart from `themeId`, it's the other most relevant arg here and also quite present in the other options (see all the `hideForSite` and `hideForTheme` fields).